C/c4Database.cc

```diff
@@ -153,7 +153,12 @@ static C4DatabaseConfig newToOldConfig(const C4DatabaseConfig2 &config2) {
 
 /*static*/ void C4Database::copyNamed(slice sourcePath, slice destinationName, const Config &config) {
     ensureConfigDirExists(config);
-    FilePath from(sourcePath);
+    std::string sourcePathString = sourcePath.asString();
+        if (sourcePathString.back() != '/')
+        {
+            sourcePathString.append("/");
+        }
+    FilePath from((slice(sourcePathString)));
```

Comment:
I think the real problem is in FilePath. Currently, FilePath::isDir() is based on syntax, whether there is a trailing slash. This can be a problem other place. /path/to_db, whether it ends with slash better not to be significant. 
For Apple, for example, we pass it to Apple's API that has RECURSIVE option. In Linux, we currently implement our own RECURSIVE based on the syntax of the source path. In my opinion, we'd better make our recursive copy more robust.

Reply:
From what I found, FilePath `_dir` is enforced to end with separator `/`.

Reply:
yes, that's at the syntactic level. But at the time of execution, we may adjust it based on the actual facts. For example, we pass fromPathSrc to Apple API, which has COPYFILE_RECURSIVE flag.

Reply:
> I think the real problem is in FilePath. Currently, FilePath::isDir() is based on syntax, whether there is a trailing slash.

With the fact that a FilePath object can be created from a non-existing path, probably the problem is how to use the current FilePath the right way?

Reply:
From another angle, may this problem can be solved by adding `FilePath(std::string_view path, bool isDir)` so that it's clear that the path is dir or file.